### DISCUSSION TITLE
Add a OLED Version

### DISCUSSION THREAD

**Discussion by dersch81** [dokuhn/openDTU-BreakoutBoard#18] [Ideas]
Hi, since OpenDTU is now supporting displays I suggest to add an additional Display version of the Case and PCB to have the connector breakouts as well as mounting points with a cut out of the case lid. 

I would support a cheap an common one like this: https://www.amazon.de/dp/B01L9GC470/ref=pe_27091401_487024491_TE_item

**Comment by dersch81** [comment]
Ok, i have noticed the case lid with space for the OLED: https://www.thingiverse.com/thing:5866231

So why not adding it here? And just modify the PCB to have a Pinout to connect the OLED?

**Comment by huckes** [comment]
Hi, please find the updated version here: https://www.thingiverse.com/thing:5866226.

I also remixed the case, as the ESP-boards I had in stock are shorter and more slim (48 x 26 mm) as the boards for which the case originally was designed. 

As I had not yet stated a PR anywhere in Github, I hope someone will be so kind to state the request with the following information, thanks:

1. Add  soldering points for a connector for an OLED-display with four connection for +3.3V, GND, SDA, SCL, I would place them between the ESP and the NRF. 
2. Add soldering points for +5V and GND for use to connect a power jack instead of powering the device by the Micro-USB.
3. On the left side of the ESP-rows add annother row, so the spacing between the left and right row can also be used by Boards with 9/10 inch spacing too.
4. Provide a pin_mapping.json with the content as proposed under 'Documentation'.

Please see the attached pictures, which might explain the proposed changes better then my rusted English.

OLED-device connected 'on-the-fly'). Soldering points neccessary for 3.3V, GND, SDA (GPIO21), SCL (GPIO22).

Only 9/10 inch spacing, so the chip-posts have to be soldered inclined:
![IMG_20230224_145220621](https://user-images.githubusercontent.com/35926857/221257276-38660a6a-dffe-4b67-ba51-a631c3ff8df7.jpg)

Additional soldering points for alternative power supply connector.
![IMG_20230224_151125985](https://user-images.githubusercontent.com/35926857/221258362-402eba29-5208-4f1a-aece-578523f4345d.jpg)

**Comment by SteffMUC** [comment]
Just fyi, spent some time to add the display connection directly on the PCB, case is currently in preparation, however not yet perfectly fitting, nevertheless you can see the intermediate status and concept here:
https://github.com/SteffMUC/openDTU_wDisplay2

**Comment by thomic** [comment]
This is something the SMT (or maybe even the breakout variant) of the board can feature in a future edition. At the moment the focus is a bit on getting the SMT version done and production ready, just to get boards to the people. Still it should be possible to add a display to the board later if you got some time, soldering skills and interest. But for now -as time for pet projects is limited- I guess OLED by default will not be there in the first round.
